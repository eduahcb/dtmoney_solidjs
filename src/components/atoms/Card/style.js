import { styled } from 'solid-styled-components'

export const Root = styled.div`
  background-color: var(--shape-principal);
  padding: 24px 32px;
  border-radius: 4px;
`
