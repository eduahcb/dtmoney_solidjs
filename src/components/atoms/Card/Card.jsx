
import { splitProps } from 'solid-js'

import { Root } from './style'

export const Card = (props) => {
  const [{ children }, rest] = splitProps(props, ['children'])

  return (
    <Root {...rest}>{children}</Root>
  )
}
