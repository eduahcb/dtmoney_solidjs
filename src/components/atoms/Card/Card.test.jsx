import { describe, test, expect } from 'vitest'

import { render, screen } from '@solidjs/testing-library'
import { Card } from './Card'

describe('Card component', () => {
  test('renders children', () => {
    render(() => <Card><p>this is a text</p></Card>)

    expect(screen.getByText('this is a text')).toBeInTheDocument()
  })

  test('to match snapshot', () => {
    const { asFragment } = render(() => <Card><p>this is a text</p></Card>)

    expect(asFragment()).toMatchSnapshot()
  })
})
