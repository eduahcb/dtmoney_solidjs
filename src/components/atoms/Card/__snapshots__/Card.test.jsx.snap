// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Card component > to match snapshot 1`] = `"<div class=\\"go425013107\\"><p>this is a text</p></div>"`;
